kernel/src/syscalls.c

```diff
@@ -0,0 +1,86 @@
+// A function in the syscall table can have up to 3 64-bit parameters, and return both 64 and 128
+// bits. If you can make sure the ECX parameter is unused, all but the stack parameters can be used.
+
+#include "syscalls.h"
+#include "gdt.h"
+
+#define MAX_SYSCALLS 16
```

Comment:
MAX_SYSCALLS should probably represent how many syscalls there actually are because it's used check if they exists.